datafusion-examples/examples/sql_parsing.rs

```diff
@@ -0,0 +1,133 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
What do you think about calling this file `sql_dialect` rather than `sql_parser` to hint it is an example of how to implement a custom dialect?